openpype/plugins/publish/integrate_new.py

```diff
@@ -304,441 +565,245 @@ def register(self, instance):
-                "anatomy_data": template_data,
-                "published_files": published_files
-            }
-            self.log.debug("__ representations: {}".format(representations))
-
-        # Remove old representations if there are any (before insertion of new)
-        if existing_repres:
-            repre_ids_to_remove = []
-            for repre in existing_repres:
-                repre_ids_to_remove.append(repre["_id"])
-            io.delete_many({"_id": {"$in": repre_ids_to_remove}})
-
-        for rep in instance.data["representations"]:
-            self.log.debug("__ rep: {}".format(rep))
+                template_data["udim"] = repre["udim"][0]
+            src = os.path.join(stagingdir, fname)
+            anatomy_filled = anatomy.format(template_data)
+            template_filled = anatomy_filled[template_name]["path"]
+            repre_context = template_filled.used_values
+            dst = os.path.normpath(template_filled)
+
+            # Single file transfer
+            self.log.debug("source: {}".format(src))
+            self.log.debug("destination: `{}`".format(dst))
+            repre["transfers"] = [src, dst]
+
+            repre['published_path'] = dst
+
+        if repre.get("udim"):
+            repre_context["udim"] = repre.get("udim")  # store list
+
+        for key in self.db_representation_context_keys:
+            value = template_data.get(key)
+            if not value:
+                continue
+            repre_context[key] = template_data[key]
 
-        io.insert_many(representations)
-        instance.data["published_representations"] = (
-            published_representations
-        )
-        # self.log.debug("Representation: {}".format(representations))
-        self.log.info("Registered {} items".format(len(representations)))
+        # Define representation id
+        repre_id = io.ObjectId()
 
-    def integrate(self, instance):
-        """ Move the files.
+        # Use previous representation's id if there is a name match
+        repre_name_lower = repre["name"].lower()
+        for _archived_repres in archived_repres:
+            if repre_name_lower == _archived_repres["name"].lower():
+                repre_id = _archived_repres["orig_id"]
+                break
 
-            Through `instance.data["transfers"]`
+        # todo: `repre` is not the actual `representation` entity
+        #       we should simplify/clarify difference between data above
+        #       and the actual representation entity for the database
+        data = repre.get("data") or {}
+        data.update({'path': repre["published_path"], 'template': template})
+        representation = {
+            "_id": repre_id,
+            "schema": "openpype:representation-2.0",
+            "type": "representation",
+            "parent": version["_id"],
+            "name": repre['name'],
+            "data": data,
+            "dependencies": instance.data.get("dependencies", "").split(),
```

Comment:
Can someone confirm for me that this is unused/deprecated?

I believe it originated from the RigLoader in Maya which stored ['dependencies'](https://github.com/pypeclub/OpenPype/blob/0f2ca2e0432655562739813fd4114f06870e7184/openpype/hosts/maya/plugins/load/load_reference.py#L159) on load. Oddly enough it's storing just the representation id of the actual rig that was loaded (which seems odd) - the value also didn't get updated on updates of the rigs.

It seems extra weird that the value on the instance wasn't just a list to begin with but expecting a concatenated string?

Anyway, with 'input dependencies' the source rig representation for a published alembic would already be retrievable and thus also more reliable than this old unused logic. @mkolar  Correct?